extensions/dicom-video/src/viewports/OHIFCornerstoneVideoViewport.tsx

```diff
@@ -1,26 +1,47 @@
 import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 
-function OHIFCornerstoneVideoViewport({ displaySets }) {
+function OHIFCornerstoneVideoViewport({ displaySets, servicesManager }) {
   if (displaySets && displaySets.length > 1) {
     throw new Error(
       'OHIFCornerstoneVideoViewport: only one display set is supported for dicom video right now'
     );
   }
 
-  const { videoUrl } = displaySets[0];
+  const { videoUrl, requiresAuthorization } = displaySets[0];
   const mimeType = 'video/mp4';
   const [url, setUrl] = useState(null);
+  const { userAuthenticationService } = servicesManager?.services || {};
 
   useEffect(() => {
     const load = async () => {
-      setUrl(await videoUrl);
+      let resolvedUrl = await videoUrl;
+      
+      if (requiresAuthorization && userAuthenticationService) {
+        const authHeaders = userAuthenticationService.getAuthorizationHeader();
+        
+        if (authHeaders && authHeaders.Authorization) {
+          try {
+            const response = await fetch(resolvedUrl, {
```

Comment:
If you load the URL this way, there is no good way to avoid loading the entire video before displaying it.  That will prevent any large videos from displaying at all because there isn't a way to pass in the authorization header to the video playback.  There should be a way to turn off the passing in of the authorization token or to incorporate it into the URL.